circup/__init__.py

```diff
@@ -1277,6 +1080,34 @@ def main(ctx, verbose, path, board_id, cpy_version):  # pragma: no cover
         logger.warning(ex)
 
 
+def get_device_path(host, password, path):
+    """
+    :param host Hostname or IP address.
+    :param password REST API password.
+    :param path File system path.
+    :return device URL or None if the device cannot be found.
+    """
+    if path:
+        device_path = path
+    elif host:
+        if password is None:
+            click.secho("--host needs --password", fg="red")
+            sys.exit(1)
+
+        # pylint: disable=no-member
+        # verify hostname/address
+        try:
+            socket.getaddrinfo(host, 80, proto=socket.IPPROTO_TCP)
+        except socket.gaierror:
+            click.secho("Invalid host: {}".format(host), fg="red")
+            sys.exit(1)
+        # pylint: enable=no-member
+        device_path = f"http://:{password}@" + host
```

Comment:
I'd expect the backend constructors to do this validation.

Reply:
moved to WebBackend in the latest commit.